### PR TITLE
[17.0][IMP] sale_force_invoiced: update invoice status in SOL

### PR DESCRIPTION
**Pull request by odooexperts-visioncode**
update sale order line invoice status together when updating sale order invoice status by force invoiced boolean

### CODE REVIEW

File: sale_force_invoiced/__manifest__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 {
     "name": "Sale Force Invoiced",
     "summary": "Allows to force the invoice status of the sales order to Invoiced",
-    "version": "17.0.1.0.0",
+    "version": "17.0.1.1.0",
```
Comment (pedrobaeza) [inline sale_force_invoiced/__manifest__.py]: Please don't change manually module version. It's done by the bot when merging.

File: sale_force_invoiced/model/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
 from . import sale_order
+from . import sale_order_line
```
Comment (pedrobaeza) [inline sale_force_invoiced/model/__init__.py]: You can incorporate children model overrides in the same `sale_order.py` file.

File: sale_force_invoiced/model/sale_order_line.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends("order_id.force_invoiced")
+    def _compute_invoice_status(self):
+        for line in self:
+            if line.order_id and line.order_id.force_invoiced:
+                line.update({"invoice_status": "invoiced"})
+            else:
+                return super()._compute_invoice_status()
```
Comment (pedrobaeza) [inline sale_force_invoiced/model/sale_order_line.py]: The code is incorrect, as it's returning early and doing the super for all the records.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_force_invoiced/model/sale_order.py [hunk 4]
```diff
@@ -23,3 +23,15 @@ def _compute_invoice_status(self):
...
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends("order_id.force_invoiced")
+    def _compute_invoice_status(self):
+        res = super()._compute_invoice_status()
+        self.filtered(lambda sol: sol.order_id and sol.order_id.force_invoiced).update(
+            {"invoice_status": "invoiced"}
+        )
```
Comment (pedrobaeza) [inline sale_force_invoiced/model/sale_order.py]:
```suggestion
        self.filtered(lambda sol: sol.order_id.force_invoiced).invoice_status = "invoiced"
```

**Review by pedrobaeza — COMMENTED** [review]
Please squash all the commits into one.

**Review by pedrobaeza — APPROVED** [review]

**Review by dreispt — APPROVED** [review]

**Comment by odooexperts-visioncode** [comment]
Hello @pedrobaeza,
Could you please guide me how to proceed for this?
As referred in [this commit of 17.0-ocabot-merge-pr-3090-by-dreispt-bump-patch](https://github.com/OCA/sale-workflow/commits/b4807be9cf24d1cf0aca2a0d6e2dbeb341f33b0e) the test cases are from another OCA module "sale_order_priority"
image

**Comment by pedrobaeza** [comment]
Please rebase your PR to check if the error is reproducible here. Any kind of patch (here or in the other module) should be provided for fixing the branch. You can also try a dummy PR to see if the problem is not directly related to this PR.

**Comment by odooexperts-visioncode** [comment]
Hello @pedrobaeza, I have updated the PR branch. Can you please check once?

**Comment by pedrobaeza** [comment]
You have to rebase, not to pull. Please redo it and forced push

**Comment by odooexperts-visioncode** [comment]
@pedrobaeza Done!

**Pull request merged**